sktime/annotation/hmm.py

```diff
@@ -0,0 +1,275 @@
+    belief of the probability distribution of hidden states at the
+    start of the observation sequence).
+
+    Parameters
+    ----------
+    emission_funcs : a list of functions, which should be properly
+        normalized PDFs over the same space as the observed data.
+    transition_prob_mat: a nxn array of probabilities, where 'n'
+        represents the number of hidden states in the HMM.  Each
+        row should sumn to 1 in order to be properly normalized
+        (ie the j'th column in the i'th row represents the
+        probability of transitioning from state i to state j.)
+    initial_probs: optional, a 1d array of probabilities across
+        the starting states. Should match prior beliefs.  If none
+        is passed will give each state an equal initial probability.
+
+    Examples
+    --------
+    >>> from sktime.annotation.hmm import HMM
+    >>> from scipy.stats import norm
+    >>> from numpy import asarray
+    >>> # define the emission probs for our HMM model:
+    >>> centers = [3.5,-5]
+    >>> sd = [.25 for i in centers]
+    >>> emi_funcs = [(norm.pdf, {'loc': mean,
+    ...  'scale': sd[ind]}) for ind, mean in enumerate(centers)]
+    >>> test = HMM(emi_funcs, asarray([[0.25,0.75], [0.666, 0.333]]))
+    >>> # generate synthetic data (or of course use your own!)
+    >>> obs = asarray([3.7,3.2,3.4,3.6,-5.1,-5.2,-4.9])
+    >>> test = test.fit(obs)
+    >>> labels = test.predict(obs)
+    """
+
+    # plan to update to make multivariate.
+    _tags = {"univariate-only": True, "fit_is_empty": False}
+
+    def __init__(
+        self,
+        emission_funcs: list,
+        transition_prob_mat: np.ndarray,
+        initial_probs: np.ndarray = None,
+    ):
+        self.initial_probs = initial_probs
+        self.num_states = len(emission_funcs)
+        self.emission_funcs = emission_funcs
+        self.transition_prob_mat = transition_prob_mat
+        self.states = [i for i in range(self.num_states)]
+        self._validate_init()
+        super(HMM, self).__init__(fmt="dense", labels="int_label")
+
+    def _validate_init(self):
+        """Verify that the parameters passed to init are well behaved."""
+        tran_mat_len = len(self.transition_prob_mat[:, 0])
+        # transition_prob_mat should be square:
+        if not len(self.transition_prob_mat.shape) == 2 or not tran_mat_len == len(
+            self.transition_prob_mat[0, :]
+        ):
+            raise ValueError(
+                "Transtion Probability must be 2D square, but got an"
+                f"object of size {self.transition_prob_mat.shape}"
+            )
+        # number of states should be consistent!
+        init_prob_len = self.num_states
+        if self.initial_probs is not None:
+            init_prob_len = len(self.initial_probs)
+        if not tran_mat_len == self.num_states == init_prob_len:
+            raise ValueError(
+                "Number of hidden states is inconsistent!  Was passed "
```

Comment:
say more explicitly:
* what *needs* to be the case
* which variables are being referred to here, so the user can compare to the function docstring by matching variable names rather than trying to match text